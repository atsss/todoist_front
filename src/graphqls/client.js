import { ApolloClient } from "apollo-client";
import { ApolloLink } from "apollo-link";
import { onError } from "apollo-link-error";
import { createHttpLink } from "apollo-link-http";
import { InMemoryCache } from "apollo-cache-inmemory";
import { CLIENT, UID, ACCESS_TOKEN } from "../variables/localStrageKey";
import { SERVER_ENDPOINT } from "../variables/endpoint";

const httpLink = createHttpLink({
  uri: SERVER_ENDPOINT
});

const middlewareLink = new ApolloLink((operation, forward) => {
  operation.setContext({
    headers: {
      CLIENT: localStorage.getItem(CLIENT) || null,
      UID: localStorage.getItem(UID) || null,
      ACCESS_TOKEN: localStorage.getItem(ACCESS_TOKEN) || null
    }
  });
  return forward(operation);
});

const errorLink = onError(({ networkError }) => {
  if (networkError.statusCode === 401) {
    console.log("hi");
  }
});

const link = errorLink.concat(middlewareLink.concat(httpLink));

const client = new ApolloClient({
  link,
  cache: new InMemoryCache()
});

export default client;
