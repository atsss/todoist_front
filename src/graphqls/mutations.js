import gql from "graphql-tag";

export const DONE_TASK_MUTATION = gql`
  mutation DoneTaskMutation($id: ID!) {
    doneTask(id: $id) {
      id
      name
      dueDate
    }
  }
`;

export const LOGIN_MUTATION = gql`
  mutation LoginMutation($email: String!, $password: String!) {
    signinUser(email: { email: $email, password: $password }) {
      token
    }
  }
`;
