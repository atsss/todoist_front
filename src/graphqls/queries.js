import gql from "graphql-tag";

export const GET_TASKS = gql`
  {
    tasks {
      id
      name
      dueDate
    }
  }
`;

export const GET_TASK = gql`
  query Task($id: ID!) {
    task(id: $id) {
      name
      dueDate
    }
  }
`;
