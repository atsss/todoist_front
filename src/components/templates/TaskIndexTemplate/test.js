import React from "react";
import renderer from "react-test-renderer";
import { MockedProvider } from "@apollo/react-testing";
import TaskIndexTemplate from "./";
import { CREATE_RESULT_MUTATION } from "../../../graphqls/mutations";

it("snapshot", () => {
  const tree = setupTree();
  expect(tree).toMatchSnapshot();
});

const setupTree = () =>
  renderer
    .create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <TaskIndexTemplate tasks={tasks} />
      </MockedProvider>
    )
    .toJSON();

const tasks = [
  {
    id: "1",
    name: "hoge",
    schedule: { hour: 10, minute: 30 },
    tags: [{ name: "fuga" }]
  }
];

const result = {
  id: 1,
  score: 8,
  task: { id: 1, name: "hoge" }
};

const mocks = [
  {
    request: {
      query: CREATE_RESULT_MUTATION,
      variables: { taskId: 1, score: 8 }
    },
    result: {
      data: { createResult: result }
    }
  }
];
