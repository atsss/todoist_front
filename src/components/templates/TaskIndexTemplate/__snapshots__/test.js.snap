// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <div>
    <div
      className="root "
    >
      <h2
        className="h h2 "
      >
        今日
      </h2>
      <p
        className="sub s normal u-ml5"
      >
        12月3日
      </p>
    </div>
    <ul>
      <li
        className="root "
      >
        <div
          className="block"
        >
          <p
            className="default m bold "
          >
            10:30
          </p>
          <p
            className="default m normal u-ml5"
          >
            hoge
          </p>
          <span
            className="tag is-info is-light u-ml10"
          >
            fuga
          </span>
        </div>
        <div
          className="block"
        >
          <div
            className="field u-mt10 "
          >
            <div
              className="select"
            >
              <select
                onChange={[Function]}
                value={8}
              >
                <option
                  value={0}
                >
                  0
                </option>
                <option
                  value={3}
                >
                  3
                </option>
                <option
                  value={6}
                >
                  6
                </option>
                <option
                  value={8}
                >
                  8
                </option>
                <option
                  value={10}
                >
                  10
                </option>
              </select>
            </div>
          </div>
          <button
            className="button u-ml10"
            onClick={[Function]}
          >
            <p
              className="white m bold "
            >
              done!
            </p>
          </button>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
